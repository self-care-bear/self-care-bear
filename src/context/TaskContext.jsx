import { createContext, useContext, useEffect, useState } from 'react';
import { v4 as uuid } from 'uuid';
// import { getUser } from '../services/auth';

export const TaskContext = createContext();

const TaskProvider = ({ children }) => {
  const [taskList, setTaskList] = useState([]);
  const [selectedTasks, setSelectedTasks] = useState([]);
  const value = { taskList, setTaskList, selectedTasks, setSelectedTasks };

  return <TaskContext.Provider value={value}>{children}</TaskContext.Provider>;
};

const useTasks = () => {
  const context = useContext(TaskContext);

  if (context === undefined) {
    throw new Error('To use useTasks you must wrap component in TaskProvider');
  }

  return context;
};

export { TaskProvider, useTasks };
