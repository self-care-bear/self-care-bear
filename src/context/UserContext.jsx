import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/auth';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  const value = { user, setUser };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('To use useUser you must wrap component in UserProvider');
  }

  return context;
};

export { UserProvider, useUser };
