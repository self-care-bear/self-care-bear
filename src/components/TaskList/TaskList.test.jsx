import { render } from '@testing-library/react';
import { TaskProvider } from '../../context/TaskContext';
import { UserProvider } from '../../context/UserContext';
import TaskList from './TaskList';

test('tasklist renders', () => {
  const { container } = render(
    <UserProvider>
      <TaskProvider>
        <TaskList />
      </TaskProvider>
    </UserProvider>
  );

  expect(container).toMatchSnapshot();
});
