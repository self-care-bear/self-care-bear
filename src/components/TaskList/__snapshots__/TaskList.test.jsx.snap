// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tasklist renders 1`] = `
<div>
  <div>
    TaskList
  </div>
</div>
`;
